frontend/modelarchive/src/main/java/org/pytorch/serve/archive/ModelArchive.java

```diff
@@ -59,7 +59,14 @@ public static ModelArchive downloadModel(String modelStore, String url)
                 throw new FileAlreadyExistsException(marFileName);
             }
             try {
-                FileUtils.copyURLToFile(new URL(url), modelLocation);
+                if (url.indexOf("file://") != -1) {
+                    String originalPath = url.replace("file://", "");
+                    String path = FilenameUtils.getFullPath(originalPath);
+                    File source = new File(path, marFileName);
+                    FileUtils.copyFile(source, modelLocation);
+                } else {
+                    FileUtils.copyURLToFile(new URL(url), modelLocation);
```

Comment:
NIT: Can't we just use `file:///<path>` as the url and remove the if-else here?

Reply:
> NIT: Can't we just use `file:///<path>` as the url and remove the if-else here?

@maaquib  `FileUtils.copyURLToFile` is a two step process. It downloads the file from URL and copies it to the destination. if we pass `file://<path>` as url, step one would fail.

Reply:
[This](https://repl.it/@maaquib/ReasonablePreciousMaps) seems to be working

Reply:
@maaquib Thanks. Removed the if else block. Just the pattern change is enough for accommodating file://